sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -82,47 +83,76 @@ public ServiceBusMessage(byte[] body) {
+        this.amqpAnnotatedMessage = new AmqpAnnotatedMessage(receivedMessage.getAmqpAnnotatedMessage());
         this.context = Context.NONE;
         setMessageId(receivedMessage.getMessageId());
         setScheduledEnqueueTime(receivedMessage.getScheduledEnqueueTime());
         setContentType(receivedMessage.getContentType());
         setCorrelationId(receivedMessage.getCorrelationId());
-        setLabel(receivedMessage.getLabel());
+        setSubject(receivedMessage.getLabel());
         setPartitionKey(receivedMessage.getPartitionKey());
         setReplyTo(receivedMessage.getReplyTo());
         setReplyToSessionId(receivedMessage.getReplyToSessionId());
         setTimeToLive(receivedMessage.getTimeToLive());
         setTo(receivedMessage.getTo());
         setSessionId(receivedMessage.getSessionId());
         setViaPartitionKey(receivedMessage.getViaPartitionKey());
+
+        //TODO (Hemant): Cleanup the values from AmqpAnnotatedMessage which should never be set by user.
+        // Some values of MessageAnnotations , Header.deliveryCount etc
+    }
+
+    /**
+     *
+     * @return
+     */
+    public AmqpAnnotatedMessage getAmqpAnnotatedMessage() {
+        return amqpAnnotatedMessage;
     }
 
     /**
      * Gets the set of free-form {@link ServiceBusMessage} properties which may be used for passing metadata associated
-     * with the {@link ServiceBusMessage} during Service Bus operations. A common use-case for {@code properties()} is
-     * to associate serialization hints for the {@link #getBody()} as an aid to consumers who wish to deserialize the
-     * binary data.
+     * with the {@link ServiceBusMessage} during Service Bus operations. A common use-case for
+     * {@code applicationProperties()} is to associate serialization hints for the {@link #getBody()} as an aid to
+     * consumers who wish to deserialize the binary data.
      *
      * @return Application properties associated with this {@link ServiceBusMessage}.
      */
-    public Map<String, Object> getProperties() {
-        return properties;
+    public Map<String, Object> getApplicationProperties() {
+        return amqpAnnotatedMessage.getApplicationProperties();
     }
 
     /**
      * Gets the actual payload/data wrapped by the {@link ServiceBusMessage}.
      *
      * <p>
      * If the means for deserializing the raw data is not apparent to consumers, a common technique is to make use of
-     * {@link #getProperties()} when creating the event, to associate serialization hints as an aid to consumers who
+     * {@link #getApplicationProperties()} when creating the event, to associate serialization hints as an aid to consumers who
      * wish to deserialize the binary data.
      * </p>
      *
      * @return A byte array representing the data.
      */
     public byte[] getBody() {
-        return Arrays.copyOf(body, body.length);
+        byte[] body = null;
+        switch(amqpAnnotatedMessage.getBody().getBodyType()) {
```

Comment:
Why not assign (amqpMessage.getBody().getBodyType()) to a local variable? This will make it easier to read.